driver/src/main/java/org/kaazing/k3po/driver/internal/Robot.java

```diff
@@ -218,32 +200,41 @@ public String getObservedScript() {
         return (progress != null) ? progress.getObservedScript() : null;
     }
 
-    public boolean isDestroyed() {
-        return destroyed;
-    }
+    public ChannelFuture dispose() {
+        if (preparedFuture == null) {
+            // no need to clean up if never started
+            disposedFuture.setSuccess();
+        } else if (!disposedFuture.isDone()) {
+            ChannelFuture future = abort();
+            future.addListener(new ChannelFutureListener() {
 
-    public boolean destroy() {
-
-        if (destroyed) {
-            return true;
-        }
-
-        abort();
+                @Override
+                public void operationComplete(ChannelFuture future) throws Exception {
+                    // close server and client channels
+                    // final ChannelGroupFuture closeFuture =
+                    serverChannels.close().addListener(new ChannelGroupFutureListener() {
+
+                        @Override
+                        public void operationComplete(ChannelGroupFuture future) throws Exception {
+                            clientChannels.close();
+                            try {
+                                bootstrapFactory.shutdown();
+                                bootstrapFactory.releaseExternalResources();
+                            } catch (Exception e) {
+                                if (LOGGER.isDebugEnabled()) {
+                                    LOGGER.error("Caught exception releasing resources", e);
+                                }
```

Comment:
Do you think the `disposedFuture` should fail here? What would that imply for the caller?